src/plugins/sqlite.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+const path = require("path");
+const fsextra = require("fs-extra");
+const sqlite3 = require("sqlite3");
+const Helper = require("../helper");
+const Msg = require("../models/msg");
+
+const currentSchemaVersion = 1;
```

Comment:
At first I was thinking of storing the version of The Lounge in the schema, but that would be pretty terrible when running and updating from `master`.

How are we planning to do migrations?
First versions of Rails were storing migration versions as integer like this, but they ended with so many problems that they switched to timestamp and have kept it that way since. I'm no fan of Rails (it's part of my day job), but their migration system is pretty nicely done. I'm expecting candidates in the Node world, do we want to look at this before shipping this feature?

Reply:
Timestamp is still an integer, so I don't see how that solves anything.

Reply:
It's not about the type, it's about the value. Each migration id is a timestamp, so there is no conflicts across PRs, order is easier to manage, etc.

But really it doesn't matter that much, because I would _much_ prefer if we don't reinvent the migration wheel and use something off-the-shelf, which would enforce its conventions.

Reply:
I'll change `1` to a timestamp. I thought about migrations more, and an existing solution that merely executes SQL statements from a file will not work for us in the future.

Because we store JSON objects, we will surely need to change the structure of it in the future (e.g. when `from` was changed into an object of `{nick, mode}`), which would require iteration in javascript as it would be very impractical trying to write a SQL query to manipulate the JSON.

It's hard to predict what kind of migrations we will have just yet, it's hard to design for it. Good thing is that we're not limited by our implementation until the need arises.

Reply:
> Each migration id is a timestamp, so there is no conflicts across PRs, order is easier to manage, etc.

I mean, there will be a conflict either way because changing the same line in a file will create a git conflict. But either way, this is what we want, because even if the version is a timestamp, it'll need to change if someone else has merged since I started my development, because otherwise my time might be behind master now, and it looks like the schema is in the past.

Reply:
👍 to both your comments.

@xPaw, I agree that SQL migrations don't work for us. Rails (I'm just taking this as an example because that's my only experience with migrations, not because I want what they're doing specifically) does migrations in Ruby, and it's very useful to migrate data along with schema.
I can see how JavaScript migrations would be a much, much better solution for us for precisely what you're saying. It's indeed likely our migrations will more often be about the stored JSON than the SQL schema, and JS would give us that for free. [There are many articles on the subject](https://kostasbariotis.com/data-migration-with-nodejs/) for when we need that.
Or... don't store JSON and store things in traditional SQL tables/columns 😅

@McInkay, yes re: conflict, but I meant when multiple PRs do changes to the schema, they have to compete on the number, which is why Rails moved away from that and went for timestamps. And you're right about conflicting in the file itself and wanting this on purpose, [in fact this is exactly what Rails does and for that very reason](http://edgeguides.rubyonrails.org/active_record_migrations.html#schema-dumps-and-source-control).